plugins/system/schemaorg/src/Extension/Schemaorg.php

```diff
@@ -462,6 +473,68 @@ public function onBeforeCompileHead()
         }
     }
 
+    /**
+     * Clean the schema and remove empty fields
+     *
+     * @param   array  $schema
+     *
+     * @return  array
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    private function cleanupSchema($schema)
+    {
+        $result = [];
+
+        foreach ($schema as $key => $value) {
+            if (is_array($value)) {
+                // Subtypes need special handling
+                if (!empty($value['@type'])) {
+                    if ($value['@type'] === 'ImageObject') {
+                        if (!empty($value['url'])) {
+                            $value['url'] = $this->prepareImage($value['url']);
+                        }
+
+                        if (empty($value['url'])) {
+                            $value = [];
+                        }
+                    } elseif ($value['@type'] === 'Date') {
+                        if (!empty($value['value'])) {
+                            $value['value'] = $this->prepareDate($value['value']);
+                        }
+
+                        if (empty($value['value'])) {
+                            $value = [];
+                        }
+                    }
+
+                    // Go into the array
+                    $value = $this->cleanupSchema($value);
+
+                    // We don't save when the array contains only the @type
+                    if (empty($value) || count($value) <= 1) {
+                        $value = null;
+                    }
+                } elseif ($key == 'genericField') {
+                    foreach ($value as $field) {
+                        $result[$field['genericTitle']] = $field['genericValue'];
+                    }
+
+                    continue;
+                }
+            }
+
+            // No data, no pary
```

Comment:
```suggestion
            // No data, no play
```

I think thats what you meant?